File: laba2/task5.cs
using System;
using System.Text;

public class PrimeDecomp
{
    public static string factors(int lst)
    {
        StringBuilder result = new StringBuilder();
        int n = lst;
        
        for (int i = 2; i <= n; i++)
        {
            int count = 0;
            while (n % i == 0)
            {
                count++;
                n /= i;
            }
            
            if (count > 0)
            {
                if (count == 1)
                    result.Append($"({i})");
                else
                    result.Append($"({i}**{count})");
            }
        }
        
        return result.ToString();
    }
}

File: laba2/task5.js
function primeFactors(n) {
  let result = "";
  for (let i = 2; i * i <= n; i++) {
    let count = 0;
    while (n % i === 0) {
      count++;
      n /= i;
    }
    if (count > 0) result += "(" + i + (count > 1 ? "**" + count : "") + ")";
  }
  if (n > 1) result += "(" + n + ")";
  return result;
}
